src/app/core/internal/impl/CircuitInternal.ts

```diff
@@ -111,40 +123,30 @@ export class CircuitInternal {
                 newPorts?.forEach(addPort);
                 if (op.inverted)
                     op.deadWires?.forEach(addWire);
-                break;
+                return OkVoid();
             }
             case "ConnectWireOp": {
-                const port1 = this.getPortByID(op.w.p1);
-                const port2 = this.getPortByID(op.w.p2);
-                if (!port1 || !port2)
-                    throw new Error("Provided port GUIDs are invalid");
-
-                checkWireConnectivity(port1, port2);
-                checkWireConnectivity(port2, port1);
-
-                (op.inverted ? deleteWire : addWire)(op.w);
-                break;
+                return this.getPortByID(op.w.p1)
+                    .andThen((port1) => this.getPortByID(op.w.p2)
+                        .andThen((port2) => checkWireConnectivity(port1, port2)
+                            .andThen((_) => checkWireConnectivity(port2, port1)
+                                .map((_) => { (op.inverted ? deleteWire : addWire)(op.w); }))));
```

Comment:
maybe consider a `.and` method

Reply:
Played around with a method like:
```ts
public andBind<U>(u: Result<U, E>): Result<[T, U], E>
```
but I don't think the current state of things warrants extending `Result`.  Punting this until we have cases that would benefit from this more (`N>=4`)